lumicks/pylake/kymo.py

```diff
@@ -181,7 +181,9 @@ def _plot(self, image, **kwargs):
         import matplotlib.pyplot as plt
 
         width_um = self._ordered_axes()[0]["scan width (um)"]
-        duration = (self.stop - self.start) / 1e9
+        # Compute average pixel time along kymo line (time at the center of the scanline)
+        mean_times = np.mean(self.timestamps, axis=0).astype(np.int64)
+        duration = (mean_times[-1] - mean_times[0]) / 1e9
```

Comment:
I think that the `mean()` is redundant work here since we only care about the difference. This could be simplified to:
```py
        ts = self.timestamps
        duration = (ts[0, -1] - ts[0, 0]) / 1e9
```